File: moddb-app/models/requests/mods/SearchModsRequest.ts
import type { ModSortDirection, ModSortType } from '~/store/mods';

export interface SearchModsRequest {
  text?: string;
  sort?: ModSortType;
  direction?: ModSortDirection;
  author?: string;
  side?: string;
  gameVersion?: string;
  tags?: string[];
}

File: VsModDb/Models/Requests/Mods/SearchModsRequest.cs
using VsModDb.Models.Mods;

namespace VsModDb.Models.Requests.Mods;

public class SearchModsRequest
{
    public string? Text { get; init; }
    public ModSortType? Sort { get; init; }
    public ModSortDirection? Direction { get; init; }
    public string? Author { get; init; }
    public string? Side { get; init; }
    public string? GameVersion { get; init; }
    public List<string>? Tags { get; init; }
    public int? Skip { get; init; }
    public int? Take { get; init; }
}